### PR TITLE
Rewrite most of the handler, tests were introduced

### PR DESCRIPTION
**Pull request by roiravhon**

### CODE REVIEW

**Comment by roiravhon** [comment]
Tests are passing for 2.7
Starting 3.X adjustments (and 2.6 if it will work, as a best effort)

File: tests/mockLogzioListener/listener.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
+# noinspection PyUnresolvedReferences
```
Comment (amirkalron) [inline tests/mockLogzioListener/listener.py]: you can use ListenerHandlerTest(unittest.TestCase):
https://docs.python.org/2/library/unittest.html
integrates nice with IDE,maybe jenkins

File: tests/mockLogzioListener/persistentFlags.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
+# Modules imported only once, this is effectively a singleton
+
+
+class PersistentFlags:
```
Comment (amirkalron) [inline tests/mockLogzioListener/persistentFlags.py]: saw this,a nice python palavra
@property
https://www.programiz.com/python-programming/property

File: tests/mockLogzioListener/logsList.py [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# Modules imported only once, this is effectively a singleton
```
Comment (amirkalron) [inline tests/mockLogzioListener/logsList.py]: if only holds a list of logs,maybe redundant

File: tests/mockLogzioListener/listener.py [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+        self.logs_list = []
+
+    def set_server_error(self):
+        self.persistent_flags.set_server_error()
+
+    def clear_server_error(self):
+        self.persistent_flags.clear_server_error()
+
+
+def find_available_port():
```
Comment (amirkalron) [inline tests/mockLogzioListener/listener.py]: _find_available_port

File: tests/mockLogzioListener/persistentFlags.py (hunk 2 above)
Comment (roiravhon) [inline tests/mockLogzioListener/persistentFlags.py]: Cool, didn't know about this

File: tests/mockLogzioListener/logsList.py (hunk 3 above)
Comment (roiravhon) [inline tests/mockLogzioListener/logsList.py]: Its being accessed from 2 different classes that I can't pass this as a parameter.
So I'm using the "import once" policy of python as a singleton. that is the only purpose

**Pull request merged**